src/kernel-install/kernel-install

```diff
@@ -85,13 +85,14 @@ fi
 KERNEL_VERSION="$1"
 KERNEL_IMAGE="$2"
 
-# Reuse directory created without a machine ID present if it exists.
-if [[ -d /efi/Default ]] || [[ -d /boot/Default ]] || [[ -d /boot/efi/Default ]]; then
-    MACHINE_ID="Default"
-elif [[ -f /etc/machine-id ]]; then
-    read MACHINE_ID < /etc/machine-id
-else
-    MACHINE_ID="Default"
+if [ -z "$MACHINE_ID" ]; then
+    # Reuse directory created without a machine ID present if it exists.
+    if [[ -d /efi/Default ]] || [[ -d /boot/Default ]] || [[ -d /boot/efi/Default ]]; then
```

Comment:
this is a bit recursive now. here you look for $BOOT basically, identifying it via existance of "Default", and then further down you look for $BOOT identifying it via $MACHINE_ID.

but i think the logic here is inverted: if the machine ID dir already exists we should always use that, and use Default only if it doesn't.

i.e. load the proper machine ID first if we can. Search $BOOT_ROOT with it and use it if found, and we are done. if that didn't work, then search $BOOT_ROOT via "Default", and if we find it use that and we are done. otherwise, go back to machine id based stuff...

Reply:
Updated, see how it strikes ya